src/server/cea_608_encoder/byte_pair_generator.py

```diff
@@ -76,6 +77,10 @@ def consume_scenes(scene_list: list) -> list:
 
         if 'scene_id' not in scene:
             raise ValueError('Every scene must have a scene ID.')
+        else if scene['scene_id'] in scene_id_check:
```

Comment:
This is a good start but it's very inefficient as you have to scan this `scene_id_check` list potentially tons of times if we have lots of scenes.

What if we create a function `validate_scene_ids(scene_list)` that loops through each scene and adds scene_id along with how many times it has been seen into a dictionary: `dict = {'scene_id': 'times_it_has_been_seen'}` (this is just pseudo code) and then when we are done adding them, we loop through the dictionary looking for a `times_it_has_been_seen` that is > 1. We can also use this information to check which `scene_names` are conflicting and return that to the user for it to be easier to debug.

Reply:
Ok, I wrote validation functions instead. Let me know what you think.